From 7faa882e42d25c024acf06c6a9b883e8a83b99b2 Mon Sep 17 00:00:00 2001
From: "Alang.Huang" <Huang.Alang@inventec.com>
Date: Wed, 6 Jan 2021 02:45:33 +0000
Subject: [PATCH] change default setting: disable i2c_whitelist_check as
 default

---
 configure.ac | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index 29f58e2..b753fc0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -37,14 +37,11 @@ AS_IF([test "x$enable_boot_flag_safe_mode_support" != "xno"],
 )
 
 # Add an option to enable/disable i2c master write read command white list checking
+#change default setting: disable i2c_whitelist_check
 AC_ARG_ENABLE([i2c-whitelist-check],
-    AS_HELP_STRING([--disable-i2c-whitelist-check], [Disable I2C master write read command white list check. [default=enable]])
-)
-AS_IF([test "x$enable_i2c_whitelist_check" != "xno"],
-    AC_MSG_NOTICE([Enabling I2C master write read command white list check])
-    [cpp_flags="-DENABLE_I2C_WHITELIST_CHECK"]
-    AC_SUBST([CPPFLAGS], [$cpp_flags]),
-    AC_MSG_WARN([Disabling I2C master write read command white list check])
+    AS_HELP_STRING([--enable-i2c-whitelist-check], [Ensable I2C master write read command white list check. [default=disable]]),
+    [cpp_flags="-DENABLE_I2C_WHITELIST_CHECK"],
+    [cpp_flags=""]
 )
 
 # softoff dir specific ones
-- 
2.7.4

