From e952f8be72134f0345f5013e0bc4aadc27f21960 Mon Sep 17 00:00:00 2001
From: "pj.chen" <chen.pj@inventec.com>
Date: Thu, 21 Jan 2021 07:13:17 +0000
Subject: [PATCH] - Add hook to mACAddress to call mac_util

---
 ethernet_interface.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ethernet_interface.cpp b/ethernet_interface.cpp
index 0c83fba..1cc4525 100644
--- a/ethernet_interface.cpp
+++ b/ethernet_interface.cpp
@@ -1027,6 +1027,11 @@ std::string EthernetInterface::mACAddress(std::string value)
         execute("/sbin/ip", "ip", "link", "set", "dev", interface.c_str(),
                 "down");
         manager.writeToConfigurationFile();
+        /* Inventec hook for mac setting */
+        if (interfaceName() == "eth0") {
+            execute("/usr/bin/mac_util", "w", value.c_str());
+
+        }
     }
 
 #ifdef HAVE_UBOOT_ENV
-- 
2.7.4

